budget/models/budget_commitment.py

```diff
@@ -115,6 +115,11 @@ class BudgetCommitment(models.Model):
         states={"draft": [("readonly", False)]},
     )
 
+    @api.onchange('template_id')
+    def _onchange_template_id(self):
+        if self.line_ids:
+            self.line_ids = [(5, 0, 0)]
```

Comment:
Reference: https://github.com/odoo/odoo/blob/85fe89ae8aa8e082b7c0f1e64ace08e0a9ff560c/odoo/fields.py#L3985

```python
from odoo import Command
```

```suggestion
            self.line_ids = [Command.clear()]
```